app/presenters/auction_status_presenter_factory.rb

```diff
@@ -55,10 +72,86 @@ def end_date
   end
 
   def sign_in_link
-    link_to 'Sign in', sign_in_path
+    Url.new(link_text: 'Sign in', path_name: 'sign_in')
   end
 
   def sign_up_link
-    link_to 'sign up', sign_up_path
+    Url.new(link_text: 'sign up', path_name: 'sign_up')
+  end
+end
+
+class FutureUserIsGuest
+  attr_reader :auction
+
+  def initialize(auction:)
+    @auction = auction
+  end
+
+  def to_s
+    "This auction starts on #{start_date}. #{sign_in_link} or
+    #{sign_up_link} with GitHub to bid."
+  end
+
+  private
+
+  def start_date
+    DcTimePresenter.convert_and_format(auction.started_at)
+  end
+
+  def sign_in_link
+    Url.new(link_text: 'Sign in', path_name: 'sign_in')
+  end
+
+  def sign_up_link
+    Url.new(link_text: 'sign up', path_name: 'sign_up')
+  end
+end
+
+class FutureUserIsVendor
```

Comment:
Personally, I'm not so hot for implicit duck typing like this because I feel it makes it harder for a new project member to get up to speed. So e.g. here, I'd mark these classes in some way to show the author's intent that they're Presenters. I bet this makes it harder for tools (Atom,Rubymine) to help as well.

Maybe I'd add the word Presenter to the class name of each. Or, I'd create a Presenter class which they'd subclass.